lib/reverse_sentence.rb

```diff
@@ -1,6 +1,65 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: The time complexity of this method is O(nm) where n is the length of my_sentence and m is the length of a word in my_sentence. The first while loop has a time complexity of O(1/2n) where n is the length of my_sentence. The second while loop has a time complexity of O(n)where n is the length of my_sentence and a nested while loop with a time complexity of (1/2m) where m is the length of the word within the sentence. So the total time complexity is O(3/2n * 1/2m) or O(nm).
+
+# Space complexity: The space complexity of this method is O(1) because the space that the variables take up stays constant regardless of the lengthe of the string.
+
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  
+  return nil if my_sentence == nil
+  
+  # reverses all characters in sentence
+  i = 0
+  j = my_sentence.length - 1
+  
+  while i < my_sentence.length / 2
+    temp_i = my_sentence[i]
+    temp_j = my_sentence[j]
+    
+    my_sentence[i] = temp_j
+    my_sentence[j] = temp_i
+    
+    i += 1 
+    j -= 1
+  end
```

Comment:
Notice that you're reversing things twice in this method.  Could you dry things up by making a helper method?